catalog/transformer/catalog-transformer-zip/src/main/java/org/codice/ddf/catalog/transformer/zip/ZipCompression.java

```diff
@@ -84,81 +85,76 @@ public BinaryContent transform(
-              zipParameters.setSourceExternalStream(true);
-              zipParameters.setFileNameInZip(METACARD_PATH + metacard.getId());
+  private void createZip(SourceResponse upstreamResponse, String filePath)
+      throws CatalogTransformerException {
+    try (FileOutputStream fileOutputStream = new FileOutputStream(filePath);
+        ZipOutputStream zipOutputStream = new ZipOutputStream(fileOutputStream)) {
 
-              try (ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
-                  ObjectOutputStream objectOutputStream =
-                      new ObjectOutputStream(byteArrayOutputStream)) {
+      List<Result> resultList = upstreamResponse.getResults();
+      Map<String, Resource> resourceMap = new HashMap<>();
 
-                objectOutputStream.writeObject(new MetacardImpl(metacard));
-                InputStream inputStream =
-                    new ByteArrayInputStream(byteArrayOutputStream.toByteArray());
-                zipFile.addStream(inputStream, zipParameters);
+      // write the metacards to the zip
+      resultList
+          .stream()
+          .map(Result::getMetacard)
+          .forEach(
+              metacard -> {
+                writeMetacardToZip(zipOutputStream, metacard);
 
                 if (hasLocalResources(metacard)) {
                   resourceMap.putAll(getAllMetacardContent(metacard));
                 }
+              });
 
-              } catch (IOException | ZipException e) {
-                LOGGER.debug("Failed to add metacard with id {}.", metacard.getId(), e);
-              }
-            });
-
-    resourceMap.forEach(
-        (filename, resource) -> {
-          try {
-            ZipParameters zipParameters = new ZipParameters();
-            zipParameters.setSourceExternalStream(true);
-            zipParameters.setFileNameInZip(filename);
-            zipFile.addStream(resource.getInputStream(), zipParameters);
-          } catch (ZipException e) {
-            LOGGER.debug("Failed to add resource with id {} to zip.", resource.getName(), e);
-          }
-        });
+      // write the resources to the zip
+      resourceMap.forEach(
+          (filename, resource) -> writeResourceToZip(zipOutputStream, filename, resource));
 
-    BinaryContent binaryContent;
-    try {
-      InputStream fileInputStream = new ZipInputStream(new FileInputStream(zipFile.getFile()));
-      binaryContent = new BinaryContentImpl(fileInputStream);
-      jarSigner.signJar(
-          zipFile.getFile(),
-          System.getProperty(SystemBaseUrl.EXTERNAL_HOST),
-          System.getProperty("javax.net.ssl.keyStorePassword"),
-          System.getProperty("javax.net.ssl.keyStore"),
-          System.getProperty("javax.net.ssl.keyStorePassword"));
-    } catch (FileNotFoundException e) {
-      throw new CatalogTransformerException("Unable to get ZIP file from ZipInputStream.", e);
+    } catch (IOException e) {
+      LOGGER.debug(
```

Comment:
✏️ There's really no value in debugging the same thing that you're throwing here. We might want to hide the internals from the user (although as a command line utility, I'm not sure there's much value in that) and not throw the internal IOException that occurred, only debug logging it; otherwise, just the throw should be sufficient.